server/src/Irony.cpp

```diff
@@ -258,17 +259,28 @@ void Irony::complete(const std::string &file,
       unsigned annotationStart = 0;
       bool typedTextSet = false;
 
-      if (availability == CXAvailability_NotAccessible ||
-          availability == CXAvailability_NotAvailable) {
-        continue;
-      }
-
       typedtext.clear();
       brief.clear();
       resultType.clear();
       prototype.clear();
       postCompCar.clear();
       postCompCdr.clear();
+      access.clear();
+
+      // https://github.com/Sarcasm/irony-mode/issues/367
+      switch (availability) {
+      case CXAvailability_Available:
+        access = "available";
+        break;
+      case CXAvailability_Deprecated:
+        access = "deprecated";
+        break;
+      case CXAvailability_NotAvailable:
```

Comment:
Can you put this special case first or last, with a newline to separate from the others?